src/view/components/AlphaPickerContainer.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { FunctionComponent, useCallback, useEffect, useRef, useState } from 'react';
+import AlphaPicker from '../../components/alphaPicker/alphaPicker';
+import { IQuery } from './type';
+
+type AlphaPickerProps = {
+    query: IQuery;
+    reloadItems: () => void;
+};
+
+const AlphaPickerContainer: FunctionComponent<AlphaPickerProps> = ({ query, reloadItems }: AlphaPickerProps) => {
+    const [ alphaPicker, setAlphaPicker ] = useState<AlphaPicker>();
+    const element = useRef<HTMLDivElement>(null);
+
+    alphaPicker?.updateControls(query);
+
+    const onAlphaPickerChange = useCallback((e) => {
+        const newValue = (e as CustomEvent).detail.value;
+        if (newValue === '#') {
+            query.NameLessThan = 'A';
+            delete query.NameStartsWith;
+        } else {
+            query.NameStartsWith = newValue;
+            delete query.NameLessThan;
+        }
+        query.StartIndex = 0;
+        reloadItems();
+    }, [query, reloadItems]);
+
+    useEffect(() => {
```

Comment:
on each change, you are not unmounting the useEffect, removing the event listener in "alphavaluechanged".

If this is invoked 100 times, there will be 100 event listeners in the element.

also if the reference element is used to grab the div with the class 'alphaPicker', you can pass the ref to that dom element.